tests/test_copy.py

```diff
@@ -34,4 +37,271 @@ def test_copy_dir(self):
 
         self.assertTrue(dst_fs.isdir('bar'))
         self.assertTrue(dst_fs.isdir('empty'))
-        self.assertTrue(dst_fs.isfile('bar/baz.txt'))
\ No newline at end of file
+        self.assertTrue(dst_fs.isfile('bar/baz.txt'))
+
+    def _create_sandbox_dir(self, prefix='pyfilesystem2_sandbox_', home=None):
+        if home is None:
+            return tempfile.mkdtemp(prefix=prefix)
+        else:
+            sandbox_path = os.path.join(home, prefix)
+            os.makedirs(sandbox_path, exist_ok=True)
+            return sandbox_path
+
+    def _touch(self, root, filepath):
+        #create abs filename
+        abs_filepath = os.path.join(root, filepath)
+        #ensure dir exists, caution: exist_ok works only for python>=3.2
+        #os.makedirs(os.path.dirname(abs_filepath), exist_ok=True)
+        dirname = os.path.dirname(abs_filepath)
+        try:
+            os.makedirs(dirname)
+        except OSError:
+            if os.path.exists(dirname):
+                pass
+            else:
+                raise
+        #touch file
+        with open(abs_filepath, 'a'):
+            os.utime(abs_filepath, None) #update the mtime in case the file exists, same as touch
+        return abs_filepath
+
+    def _write_file(self, filepath, write_chars=1024):
+        with open(filepath, 'w') as f:
+            f.write('1' * write_chars)
+        return filepath
+
+    def test_copy_file_if_newer_dst_older(self):
+        try:
+            #create first dst ==> dst is older the src ==> file should be copied
+            dst_dir = self._create_sandbox_dir()
+            dst_file1 = self._touch(dst_dir, "file1.txt")
+            self._write_file(dst_file1)
+
+            time.sleep(1) #sleep 1 sec to ensure dst_file1 is older
```

Comment:
I add sleeps because on trevis, tests failed in unpredictive way. For some version of python some test falied once, then the second time they succced. This is strange... so i decided to add a sleep in case was a race on file object. But probably it is not, because the behaivoe is the same with or without sleep.

I will remove the sleep, still some test will fail, do you have an idea why?
Note that py3.5 tests are failing. On my workstation i use py35 on windows and tests are all green!